cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -554,206 +714,75 @@ data GenesisParameters =
+  -> Either ProtocolParametersError (ProtocolParameters era)
+fromShelleyPParams sbe pparams =
+  let pp = case sbe of
+             ShelleyBasedEraShelley -> toProtocolParamsShelley pparams
+             ShelleyBasedEraAllegra -> toProtocolParamsShelley pparams
+             ShelleyBasedEraMary -> toProtocolParamsShelley pparams
+--           ShelleyBasedEraAlonzo -> toProtocolParamsAlonzo pparams
+  in do checkProtocolVersion sbe (protocolParamProtocolVersion pp)
+        return pp
+
+toProtocolParamsShelley :: Shelley.PParams ledgerera -> ProtocolParameters era
+toProtocolParamsShelley pparams =
+   ProtocolParameters
+        { protocolParamProtocolVersion     = (\(Shelley.ProtVer a b) -> (a,b))
+                                               $ Shelley._protocolVersion pparams
+        , protocolParamDecentralization    = Shelley.unitIntervalToRational $ Shelley._d pparams
+        , protocolParamExtraPraosEntropy   = fromPraosNonce $ Shelley._extraEntropy pparams
+        , protocolParamMaxBlockHeaderSize  = Shelley._maxBHSize pparams
+        , protocolParamMaxBlockBodySize    = Shelley._maxBBSize pparams
+        , protocolParamMaxTxSize           = Shelley._maxTxSize pparams
+        , protocolParamTxFeeFixed          = Shelley._minfeeB pparams
+        , protocolParamTxFeePerByte        = Shelley._minfeeA pparams
+        , protocolParamMinUTxOValue        = Just . fromShelleyLovelace $ Shelley._minUTxOValue pparams
+        , protocolParamStakeAddressDeposit = fromShelleyLovelace $ Shelley._keyDeposit pparams
+        , protocolParamStakePoolDeposit    = fromShelleyLovelace $ Shelley._poolDeposit pparams
+        , protocolParamMinPoolCost         = fromShelleyLovelace $ Shelley._minPoolCost pparams
+        , protocolParamPoolRetireMaxEpoch  = Shelley._eMax pparams
+        , protocolParamStakePoolTargetNum  = Shelley._nOpt pparams
+        , protocolParamPoolPledgeInfluence = Shelley._a0 pparams
+        , protocolParamMonetaryExpansion   = Shelley.unitIntervalToRational $ Shelley._rho pparams
+        , protocolParamTreasuryCut         = Shelley.unitIntervalToRational $ Shelley._tau pparams
+        , protocolParamUTxOCostPerByte     = Nothing
+        , protocolParamCostModels          = mempty
+        , protocolParamPrices              = Nothing
+        , protocolParamMaxTxExUnits        = Nothing
+        , protocolParamMaxBlockExUnits     = Nothing
+        , protocolParamMaxValSize          = Nothing
+        }
+
+
+checkProtocolVersion
+  :: ShelleyBasedEra era
+  -> (Natural, Natural)
+  -> Either ProtocolParametersError ()
+checkProtocolVersion sbe protoVer  =
+  let expectedPVer = case sbe of
+                       ShelleyBasedEraShelley -> (2,0)
+                       ShelleyBasedEraAllegra -> (3,0)
+                       ShelleyBasedEraMary -> (4,0)
+              --      ShelleyBasedEraAlonzo -> (5,0)
+  in checkCondition (protoVer == expectedPVer)
+       $ ProtocolParametersErrorWrongVersion expectedPVer protoVer
```

Comment:
And if we were to do this check, this would be the wrong check. We can (and do) have minor versions that are non-zero.

But as I said above, as far as I can see this check is entirely unnecessary. The ledger is right. It knows the protocol params. We cannot second guess them.